### DISCUSSION TITLE
テレメトリを送信および受信するコードを記述する Learn 教材を試したが受信時に ArgumentError: The connection string is missing the property: SharedAccessKeyName エラー

### DISCUSSION THREAD

**Discussion by 1ft-seabass** [1ft-seabass/ms-azure-hackathon-league-2021#41] [Show and tell]
演習 - テレメトリを送信および受信するコードを記述する - Learn | Microsoft Docs
https://docs.microsoft.com/ja-jp/learn/modules/remotely-monitor-devices-with-azure-iot-hub/4-exercise-write-code-device-telemetry?pivots=vscode-node

こちらの教材で受信時に ArgumentError: The connection string is missing the property: SharedAccessKeyName エラーが出てしまう。

**Comment by 1ft-seabass** [comment]
送信時のソースコードの connectionString 変数と同じを思ってしまい 受信時の connectionString 変数でも同じものを使ってしまうことが原因。

- 送信時
  - 個々のデバイスから送信するものなのでデバイスの接続文字列を使います
- 受信時
  - IoT Hub そのものから受信をするためポリシーの iothubowner ポリシーの接続文字列を使います

ということで別の接続文字列になります。

受信時 iothubowner ポリシーの接続文字列は IoT Hub の

- 左側のメニューで、[設定] > [共有アクセス ポリシー] を選択
- [ポリシー] の下にある iothubowner エントリをクリック
- 右側の [接続文字列 – 主キー] の内容をコピー

して取得します。

うまく設定出来れば、送信の Node.js アプリを起動したうえで、

![image](https://i.gyazo.com/251249799a860103efc7b38e7f0f1e97.png)

受信アプリを起動すると、このように受信ができます。

## 詳しい教材

詳しくは今回の教材は以下の教材の配下にある、

Azure IoT Hub を使用してデバイスをリモートで監視および制御する - Learn | Microsoft Docs
https://docs.microsoft.com/ja-jp/learn/modules/remotely-monitor-devices-with-azure-iot-hub/

こちらの、

演習 - Microsoft Azure portal を使用して、Azure IoT ハブとデバイス ID を作成する - Learn | Microsoft Docs
https://docs.microsoft.com/ja-jp/learn/modules/remotely-monitor-devices-with-azure-iot-hub/2-create-iot-hub-device-id

に書かれています。

**Reply by uecken** [reply]
取れました！ありがとうございます！
